File: src/Plotly.NET/Traces/ObjectAbstractions/Slices.fs
namespace Plotly.NET.TraceObjects

open Plotly.NET
open Plotly.NET.LayoutObjects
open DynamicObj
open System
open System.Runtime.InteropServices

type SlicesFill() =
    inherit DynamicObj()

    static member init
        (
            ?Fill: float,
            ?Locations: seq<#IConvertible>,
            ?Show: bool
        ) =
        SlicesFill() |> SlicesFill.style (?Fill = Fill, ?Locations = Locations, ?Show = Show)

    static member style
        (
            ?Fill: float,
            ?Locations: seq<#IConvertible>,
            ?Show: bool
        ) =

        fun (slicesFill: SlicesFill) ->

            slicesFill
            |> DynObj.withOptionalProperty "fill" Fill
            |> DynObj.withOptionalProperty "locations" Locations
            |> DynObj.withOptionalProperty "show" Show


type Slices() =
    inherit DynamicObj()

    static member init
        (
            ?X: SlicesFill,
            ?Y: SlicesFill,
            ?Z: SlicesFill
        ) =

        Slices() |> Slices.style (?X = X, ?Y = Y, ?Z = Z)

    static member style
        (
            ?X: SlicesFill,
            ?Y: SlicesFill,
            ?Z: SlicesFill
        ) =
        fun (slices: Slices) ->

            slices
            |> DynObj.withOptionalProperty "x" X
            |> DynObj.withOptionalProperty "y" Y
            |> DynObj.withOptionalProperty "z" Z

File: src/Plotly.NET.CSharp/ObjectAbstractions/TraceObjects/Slices.cs

using Microsoft.FSharp.Core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Plotly.NET.CSharp;

public static partial class TraceObjects
{
    public static Plotly.NET.TraceObjects.SlicesFill InitSlicesFill<LocationsType>
        (
            Optional<double> Fill = default,
            Optional<IEnumerable<LocationsType>> Locations = default,
            Optional<bool> Show = default
        ) 
            where LocationsType: IConvertible
            =>
                Plotly.NET.TraceObjects.SlicesFill.init(
                    Fill: Fill.ToOption(),
                    Locations: Locations.ToOption(),
                    Show: Show.ToOption()
                );
    public static Plotly.NET.TraceObjects.Slices InitSlices
        (
            Optional<Plotly.NET.TraceObjects.SlicesFill> X = default,
            Optional<Plotly.NET.TraceObjects.SlicesFill> Y = default,
            Optional<Plotly.NET.TraceObjects.SlicesFill> Z = default
        ) 
            =>
                Plotly.NET.TraceObjects.Slices.init(
                    X: X.ToOption(),
                    Y: Y.ToOption(),
                    Z: Z.ToOption()
                );
}

